### PR TITLE
cpu: moved all vendor headers to CPUNAME/include/vendor/

### PR DESCRIPTION
**Pull request by haukepetersen**
The reason for this is to simplify handling of vendor headers in the future, e.g. by easier excluding them from doxygen builds etc...

### CODE REVIEW

File: doc/doxygen/riot.doxyfile [hunk 1]
```diff
@@ -818,30 +818,11 @@ EXCLUDE_SYMLINKS       = NO
...
-                         */cpu/ezr32wg/include/ezr* \
-                         */cpu/*/include/atmel/* \
-                         */cpu/k60/include/MK60D10.h \
-                         */cpu/k60/include/MK60DZ10.h \
-                         */cpu/k64f/include/MK64F12.h \
-                         */cpu/kw2x/include/MKW22D5.h \
-                         */cpu/lpc11u34/include/LPC11Uxx.h \
-                         */cpu/lpc1768/include/LPC17xx.h \
-                         */cpu/lpc*/include/core_cm*.h \
+                         */cpu/*/include/vendor/*
```
Comment (kaspar030) [inline doc/doxygen/riot.doxyfile]: Why not ```*/vendor/*```?

Comment (haukepetersen) [inline doc/doxygen/riot.doxyfile]: good point. will adapt

**Comment by kaspar030** [comment]
Could you also adapt the cppcheck filters in ```dist/tools/cppcheck/check.sh```?

**Comment by haukepetersen** [comment]
done

**Comment by haukepetersen** [comment]
fixed excludes for externc and some empty line at EOF issues.

**Comment by aabadie** [comment]
I'm globally +1 with the goal of this PR but as you have noticed, I have some opened PRs that will be impacted : #6651, #6025, #6553, #6594, #6615, #6625.
Some of them are opened for a long time now (#6025) or closed to being merged (just missing a positive review)  (#6651) .
Would it be possible to merge them first ?

**Comment by gebart** [comment]
Good idea, but what about having two separate trees, instead of polluting the include tree. 
Like this:
Cpu/stm32f4/vendor/include

Also, while I can see that having to prefix vendor/ on all vendor includes has its merits with regards to readability, it may also cause a hassle when vendor headers include other vendor headers (e.g. Atmel CPUs). 
I'm writing this from the phone so I haven't really managed to get a good overview of this PR, so please excuse me if I have missed something here.

**Comment by haukepetersen** [comment]
@aabadie: I would prefer to merge this PR first (as there are probably another 20 PRs or so, that would need to be merged first...), as right now there is the effort for the 'code quality task force'.

@gebart I think this PR is not `polluting the include tree`, IMHO it's quite the contrary by cleanly separating the vendor headers. Regarding the vendor libraries (as e.g. the Atmel files), this PR has no negative impact, as they use relative include paths and find their 'peer' files locally without even looking into the include tree...

File: doc/doxygen/riot.doxyfile [hunk 2]
```diff
@@ -818,30 +818,11 @@ EXCLUDE_SYMLINKS       = NO
...
-                         */cpu/ezr32wg/include/ezr* \
-                         */cpu/*/include/atmel/* \
-                         */cpu/k60/include/MK60D10.h \
-                         */cpu/k60/include/MK60DZ10.h \
-                         */cpu/k64f/include/MK64F12.h \
-                         */cpu/kw2x/include/MKW22D5.h \
-                         */cpu/lpc11u34/include/LPC11Uxx.h \
-                         */cpu/lpc1768/include/LPC17xx.h \
-                         */cpu/lpc*/include/core_cm*.h \
+                         */vendor/* \
```
Comment (gebart) [inline doc/doxygen/riot.doxyfile]: nice cleanup here! :+1: 
Could you move it to the correct place to keep the list sorted?

**Comment by gebart** [comment]
@haukepetersen After looking at this properly on my PC I am starting to agree that this is a good solution.
Added the major tag because this touches a lot of files across the tree.

@aabadie the rebase of other PRs touching these vendor headers should be quite uncomplicated. I agree with @haukepetersen that merging this is preferable compared to merging everything else first. There are new PRs opened all the time so it would be an infinite catching up to try to get this merged.

**Comment by aabadie** [comment]
@gebart, indeed, just tested the rebase, things works fine (even if I have to modify files afterwards but that was expected anyways).

**Comment by haukepetersen** [comment]
Addressed comment by @gebart above and squashed. Now waiting for the CI once more to make sure I didn't break anything and then we should be ready to merge this.

**Comment by kaspar030** [comment]
Murdock2 currently chokes as cppcheck takes longer than 5 minutes, which is Murdock2's default job timeout. The other CI's agree, so I'd say merge!

**Pull request merged**